src/lib/mongoBackend/mongoRegisterContext.cpp

```diff
@@ -67,8 +80,9 @@ HttpStatusCode mongoRegisterContext(RegisterContextRequest* requestP, RegisterCo
         id = OID(requestP->registrationId.get());
 
         mongoSemTake(__FUNCTION__, "findOne from RegistrationsCollection");
-        reg = connection->findOne(getRegistrationsCollectionName(tenant).c_str(), BSON("_id" << id));
+        reg = connection->findOne(getRegistrationsCollectionName(tenant).c_str(), BSON("_id" << id << "servicePath" << sPath));
```

Comment:
Ok, I think I have located the place where you takes servicePath into account in registrations.

Are you checking that servicePaths with "#" are not used here? (different from discovery or notifyContextAvailability)